File: abc/abc180/abc180_c.cs
// https://atcoder.jp/contests/abc180/submissions/17490989

using System;
using System.Collections.Generic;

class Program
{
    static void Main()
    {
        var N = long.Parse(Console.ReadLine());

        var result = new List<long>();
        for (var i = 1L; i <= Math.Sqrt(N); i++)
        {
            if (N % i == 0)
            {
                result.Add(i);
                if (N / i != i)
                {
                    result.Add(N / i);
                }
            }
        }
        result.Sort();
        foreach (var r in result)
        {
            Console.WriteLine(r);
        }
    }
}

File: abc/abc180/abc180_c.fs
// https://atcoder.jp/contests/abc180/submissions/17491386

let N = stdin.ReadLine() |> int64

let rec solver (x : int64 list) i =
    if i * i > N then x
    elif N % i = 0L && N / i <> i then solver (i :: N / i :: x) (i + 1L)
    elif N % i = 0L then solver (i :: x) (i + 1L)
    else solver x (i + 1L)

solver [] 1L
|> List.sort
|> List.iter stdout.WriteLine